addons/account_edi_ubl/models/account_edi_format_ubl.py

```diff
@@ -16,6 +16,66 @@
 class AccountEdiFormat(models.Model):
     _inherit = 'account.edi.format'
 
+    ####################################################
+    # Helpers
+    ####################################################
+
+    def _is_generic_ubl(self, filename, tree):
+        return tree.tag == '{urn:oasis:names:specification:ubl:schema:xsd:Invoice-2}Invoice'
+
+    ####################################################
+    # Hooks
+    ####################################################
+
+    def _is_ubl(self, filename, tree):
+        ''' Checks if the xml can be imported with this UBL implementation. See `_is_generic_ubl`.
+            Note that you need to override _create_invoice_from_xml_tree/_update_invoice_from_xml_tree or _import_ubl
+            if you want to customize import, otherwise generic import will be used.
+
+            TO OVERRIDE
+        '''
```

Comment:
I decided to clean this code so it's more understandable. Before, the code did more stuff but it was a spaghetti and was conceptually weird :
- Import is done on the concrete formats, so we have to override account_edi hooks checking format.code.
- With ubl, we can have some generic import methods that will work for multiple format, so methods are defined in 'abstract' edi formats (ubl & en16931) but the import has to be called on 'leaf' formats because only them have the format.code.
- It could be possible to make an import not dependent on format.code by importing on the account.edi.format model and not the records, just by overriding the import methods without checking for format.code. BUT, one potential problem is that all account_edi_format installed will be used to try to import a file, it's not possible to try to parse with a selected subset of formats (note that in the future it might be useful but it isn't as of now). ALSO this is counter-intuitive with the way the export is designed and introduce some potential confusion.

The solution I had :
It was a mix of both worlds : some stuff was done automatically by the "abstract" formats, the hooks where overriden here but new hooks were defined that needed to be overriden by the child classes to work. The improvement was not huge but the code was harder to understand.

What I did :
- I kept the helpers to import in the the ubl 'abstract' formats.
- No account_edi_format hook override in the abstract formats => nothing magic happens, everything has to be done manually in the 'leaf' edi formats (but can be done easily thanks to the helpers).